util/scale_run.sh

```diff
@@ -0,0 +1,230 @@
+if [ "${#ARGS[@]}" -le "0" ]; then
+    echo "PE_VERSION not provided"
+    echo
+    usage
+    exit 1
+fi
+PE_VERSION=${ARGS[0]}
+
+# Ensure PE_VERSION is semver
+# e.g. 2019.1.1-rc0-207-g4e47830
+# Regex from: https://regexr.com/39s32
+if ! [[ $PE_VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+((-[0-9a-zA-Z-]+)*)?$ ]]; then
+    echo "Supplied PE_VERSION is malformed"
+    echo
+    usage
+    exit 1
+fi
+
+
+##########
+# Main
+##########
+
+tune=false
+# Global settings
+export BEAKER_INSTALL_TYPE=pe
+export BEAKER_PE_DIR=http://enterprise.delivery.puppetlabs.net/archives/releases/2019.1.0
+export BEAKER_PE_VER=$PE_VERSION
+export PUPPET_GATLING_SCALE_TUNE=$tune
+# export PUPPET_GATLING_SCALE_TUNE_FORCE=true
+
+
+mkdir "$HOME/gplt"
+
+echo "========================================================================"
+echo "Testing Ref Arch 1: Monolithic deploy for 10 or fewer nodes"
+
+    export PUPPET_GATLING_SCALE_BASE_INSTANCES=100
+    export PUPPET_GATLING_SCALE_ITERATIONS=1
+    export PUPPET_GATLING_SCALE_SCENARIO="Scale.json"
+    export ABS_AWS_MOM_SIZE="m5.large"
+
+    run_type="cold"
+    ec2=$ABS_AWS_MOM_SIZE
+    test="slv-414-ref1-$ec2"
+    cmd="bundle exec rake autoscale_$run_type > \"$test-$run_type.log\""
+    if [ -z $DEBUG ]; then
+        prep_gplt "$test"
+        (bundle exec rake autoscale_$run_type > "$test-$run_type.log") &
+    else
+        (echo_env) &
+    fi
+
+
+echo "========================================================================"
+echo "Testing Ref Arch 2: Monolithic deploy for up to 4,000 nodes"
+
+    export PUPPET_GATLING_SCALE_ITERATIONS=15
+    export PUPPET_GATLING_SCALE_INCREMENT=100
+    export PUPPET_GATLING_SCALE_SCENARIO="Scale.json"
+
+    run_type="cold"
+    declare -a ec2_types=("c5.xlarge" "c5.2xlarge" "c5.4xlarge")
+    for i in {0..4}; do
+        wait
+        [[ $i = 0 ]] && task="autoscale_$run_type" || task="autoscale_provisioned_$run_type"
+        for ec2 in "${ec2_types[@]}"; do
+            if [[ $ec2 == "c5.xlarge" ]]; then
+                if [[ $tune == "false" ]]; then
+                    export PUPPET_GATLING_SCALE_BASE_INSTANCES=2000
+                else
+                    export PUPPET_GATLING_SCALE_BASE_INSTANCES=2000
+                fi
+            elif [[ $ec2 == "c5.2xlarge" ]]; then
+                if [[ $tune == "false" ]]; then
+                    export PUPPET_GATLING_SCALE_BASE_INSTANCES=3800
+                else
+                    export PUPPET_GATLING_SCALE_BASE_INSTANCES=4100
+                fi
+            elif [[ $ec2 == "c5.4xlarge" ]]; then
+                if [[ $tune == "false" ]]; then
+                    export PUPPET_GATLING_SCALE_BASE_INSTANCES=4100
+                else
+                    export PUPPET_GATLING_SCALE_BASE_INSTANCES=3800
+                fi
+            else
+                export PUPPET_GATLING_SCALE_BASE_INSTANCES=3800
+            fi
+            export ABS_AWS_MOM_SIZE="$ec2"
+            test="slv-414-ref2-$ec2-tune-$tune"
```

Comment:
prefix hardcode... and since ref2 appears to be meaningful, it should have a meaningful name...  like standard or trial...

Reply:
Added -i|--run-id option to allow the user to specify an optional prefix.